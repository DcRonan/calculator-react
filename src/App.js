const App = () => {
  return (
    <div>
      <header>Calculator - React</header>
    </div>
  );
};

export default App;
