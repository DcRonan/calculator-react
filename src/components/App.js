/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from './Navbar';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import Calculator from './Calculator';

const data = {
  total: '0',
  next: null,
  operation: null,
};

const App = () => {
  const [state, setState] = useState(data);

  const handleClick = buttonName => {
    const newState = calculate(state, buttonName);
    setState(newState);
  };

  const update = () => {
    const { total, next, operation } = state;

    if (next) return next;
    return total;
  };

  return (
    <>
      <header>
        <Navbar />
      </header>
      <main>
        <section className="calc-main">
          <div style={{ width: '60%' }}>
            <Calculator />
          </div>
          <div style={{ width: '40%' }}>
            <div className="calculator">
              <Display result={update()} />
              <ButtonPanel handleClick={handleClick} />
            </div>
          </div>
        </section>
      </main>
    </>
  );
};

export default App;
