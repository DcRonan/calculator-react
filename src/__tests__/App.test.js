import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import App from '../components/App';

describe('checks if the App component renders correctly', () => {
  test('renders calculator link', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    const linkElement = screen.getByText(/Let`s do some math!/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('should render correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should display a correct number', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    const btn = screen.getByText('5');
    expect(btn).toBeInTheDocument();
  });

  test('should receive correct value when clicking specific buttons', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('0'));
    fireEvent.click(screen.getByText('='));

    expect(screen.getByText('520')).toBeInTheDocument();
  });
});
