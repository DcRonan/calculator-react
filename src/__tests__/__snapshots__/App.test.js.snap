// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if the App component renders correctly should render correctly 1`] = `
Array [
  <header>
    <nav>
      <div>
        <div
          className="title"
        >
           Math Magicians 
        </div>
      </div>
      <div
        className="navbar"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home 
        </a>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </div>
    </nav>
  </header>,
  <main>
    <section
      className="calc-main"
    >
      <div
        style={
          Object {
            "width": "60%",
          }
        }
      >
        <div
          style={
            Object {
              "fontSize": "20px",
            }
          }
        >
          Let\`s do some math!
        </div>
      </div>
      <div
        style={
          Object {
            "width": "40%",
          }
        }
      >
        <div
          className="calculator"
        >
          <div
            className="calculator display"
          >
            <h2
              style={
                Object {
                  "padding": "40px",
                }
              }
            >
              0
            </h2>
          </div>
          <section>
            <div
              className="calculator ButtonPanel"
            >
              <div
                className="group-one group"
              >
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  AC
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  +/-
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  %
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "orange",
                    }
                  }
                  type="button"
                >
                  รท
                </button>
              </div>
              <div
                className="group-two group"
              >
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  7
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  8
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  9
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "orange",
                    }
                  }
                  type="button"
                >
                  x
                </button>
              </div>
              <div
                className="group-three group"
              >
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  4
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  5
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  6
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "orange",
                    }
                  }
                  type="button"
                >
                  -
                </button>
              </div>
              <div
                className="group-four group"
              >
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  1
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  2
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  3
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "orange",
                    }
                  }
                  type="button"
                >
                  +
                </button>
              </div>
              <div
                className="group-five group"
              >
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  0
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  .
                </button>
                <button
                  className="btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "",
                    }
                  }
                  type="button"
                >
                  =
                </button>
              </div>
            </div>
          </section>
        </div>
      </div>
    </section>
  </main>,
]
`;
